### ISSUE TITLE
Possible false negative on PLR1736

### ISSUE THREAD

**Issue by sbrugman** [astral-sh/ruff#13183]
The following example I would expect to be recognised by [PLR1736](https://docs.astral.sh/ruff/rules/unnecessary-list-index-lookup/) 

Minimal reproducible example:

```python
data = {"a": 1, "b": 2}
column_names = ["a", "b"]
for index, column_name in enumerate(column_names):
    _ = data[column_names[index]]
```

Command:
```
ruff check --isolated --select PLR1736 examples/plr.py
```

Output:

```text
All checks passed!
```

Expected output:
```text
examples/plr.py:8:8: PLR1736 [*] List index lookup in `enumerate()` loop
  |
7 | for index, column_name in enumerate(column_names):
8 |     _ = data[column_names[index]]
  |             ^^^^^^^^^^^^^^^^^^^^ PLR1736
  |
  = help: Use the loop variable directly
```

Version: ruff 0.6.3
